### PR TITLE
EXPERIMENTAL_useCheckboxTree hook

### PR DESCRIPTION
**Pull request by matheusps**
#### What is the purpose of this pull request?

🚫 Deprecate `EXPERIMENTAL_useTableBulkActions`
🚫 Deprecate `EXPERIMENTAL_useTableTreeCheckboxes`
➕ Introduce `EXPERIMENTAL_useCheckboxTree` that keeps the state of a collection of checkboxes. This replaces the two above. (it works with generics, so is highly reusable ♺ )
💅 Unify the `Headings` of `EXPERIMENTAL_Table` and `EXPERIMENTAL_TableTree`.
💅 Fix the distance of checkboxes, so they are the same.

#### How should this be manually tested?

Clone the repo and `yarn && yarn start`

#### Screenshots or example usage

Before:
Screen Shot 2019-11-21 at 16 56 30

After:
Screen Shot 2019-11-21 at 16 55 41

#### Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/EXPERIMENTAL_TableTree/README.md [hunk 1]
```diff
@@ -131,7 +130,7 @@ function Example() {
     console.log(checkedItems)
```
Comment (kevinch) [inline react/components/EXPERIMENTAL_TableTree/README.md]: Useful?

Comment (matheusps) [reply react/components/EXPERIMENTAL_TableTree/README.md]: Yeah. It is just to illustrate the `onToggle` function (i found less annoying than `alert`).

Comment (matheusps) [reply react/components/EXPERIMENTAL_TableTree/README.md]: In the upcoming `full documentation PR` I'll create more meaningful examples.

Comment (kevinch) [reply react/components/EXPERIMENTAL_TableTree/README.md]: Didn't that it was in a readme. No problem yes.

**Pull request merged**